From d0aeaff0dae4c7eb7fcb88d2d33c667364e870de Mon Sep 17 00:00:00 2001
From: Victor Toso <me@victortoso.com>
Date: Tue, 4 Sep 2018 10:51:42 +0200
Subject: [PATCH] spice-vdagent.desktop: Change autostart to WindowManager
 stage

It is possible that g_get_user_special_dir() return NULL for
G_USER_DIRECTORY_DESKTOP or G_USER_DIRECTORY_DOWNLOAD; This is a sure
thing to happen in case xdg-user-dirs hasn't been run yet.

Keeping spice-vdagent to run on Initialization stage of
X-GNOME-Autostart-Phase could trigger a race with xdg-user-dirs that
also runs on Initialization, which is the earliest stage possible.

Moving to spice-vdagent to run WindowManager stage avoids the race
with xdg-user-dirs and still runs before any graphical application.

Tested also moving to Desktop stage, which is 4th stage (out of 5) and
worked without apparent drawbacks.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1623947
Signed-off-by: Victor Toso <victortoso@redhat.com>
---
 data/spice-vdagent.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/spice-vdagent.desktop b/data/spice-vdagent.desktop
index d152c12..260dace 100644
--- a/data/spice-vdagent.desktop
+++ b/data/spice-vdagent.desktop
@@ -5,4 +5,4 @@ Exec=/usr/bin/spice-vdagent
 Terminal=false
 Type=Application
 Categories=
-X-GNOME-Autostart-Phase=Initialization
+X-GNOME-Autostart-Phase=WindowManager
-- 
2.17.1

